src/test/java/teammates/sqlui/webapi/UnpublishFeedbackSessionActionTest.java

```diff
@@ -0,0 +1,170 @@
+package teammates.sqlui.webapi;
+
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.InstructorPermissionRole;
+import teammates.common.datatransfer.InstructorPrivileges;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Const;
+import teammates.storage.sqlentity.Account;
+import teammates.storage.sqlentity.Course;
+import teammates.storage.sqlentity.FeedbackSession;
+import teammates.storage.sqlentity.Instructor;
+import teammates.ui.output.FeedbackSessionData;
+import teammates.ui.output.FeedbackSessionPublishStatus;
+import teammates.ui.webapi.EntityNotFoundException;
+import teammates.ui.webapi.JsonResult;
+import teammates.ui.webapi.UnpublishFeedbackSessionAction;
+
+/**
+ * SUT: {@link UnpublishFeedbackSessionAction}.
+ */
+public class UnpublishFeedbackSessionActionTest extends BaseActionTest<UnpublishFeedbackSessionAction> {
+
+    private Course course1;
+    private FeedbackSession feedbackSession1;
+    private FeedbackSession feedbackSession2;
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.SESSION_PUBLISH;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return DELETE;
+    }
+
+    @BeforeMethod
+    void setUp() throws InvalidParametersException, EntityDoesNotExistException {
+        course1 = generateCourse1();
+        feedbackSession1 = generateSession1InCourse(course1);
+        feedbackSession2 = generateSession2InCourse(course1);
+        Instructor instructor1 = generateInstructor1InCourse(course1);
+
+        when(mockLogic.getFeedbackSession(feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession1);
+        when(mockLogic.unpublishFeedbackSession(
+                feedbackSession1.getName(), course1.getId())).thenReturn(feedbackSession2);
+        when(mockLogic.getInstructorByGoogleId(
+                course1.getId(), instructor1.getAccount().getGoogleId())).thenReturn(instructor1);
+
+        loginAsInstructor(instructor1.getAccount().getGoogleId());
+    }
+
+    @Test
+    protected void testExecute() {
+        ______TS("Typical case");
```

Comment:
No real reason to use __TS separators in unit tests. This makes it
1. difficult to determine which tests have failed
2. makes test cases dependent on each other

Instead, we should have a separate method for each test.